tests/v2/validation/rbac/globalrolesv2/globalroles_v2_webhook_test.go

```diff
@@ -0,0 +1,315 @@
+		RoleTemplateIDs: []string{roleOwner},
+	})
+	require.NoError(grw.T(), err)
+
+	_, err = createGlobalRole(grw.client, []string{inheritedRoleTemplate.ID})
+	require.NoError(grw.T(), err)
+
+	err = rbac.DeleteClusterRoletemplate(grw.client, localcluster, inheritedRoleTemplate.ID)
+	require.Error(grw.T(), err)
+
+	expectedErrMessage := "admission webhook \"rancher.cattle.io.roletemplates.management.cattle.io\" denied the request: roletemplate \"" + inheritedRoleTemplate.ID + "\" cannot be deleted because it is inherited by globalRole(s) \"" + globalRole.Name + "\""
+
+	start := "admission webhook"
+	end := "cannot be deleted because it is inherited by globalRole(s) " + globalRole.Name
+
+	getError := getErrorMessage(err.Error(), start, end)
+
+	assert.Contains(grw.T(), expectedErrMessage, getError)
+}
+
+func (grw *GlobalRolesV2WebhookTestSuite) TestAddProjectRoleTemplateInInheritedClusterRole() {
+	log.Info("Verify adding a project role template is rejected by webhook.")
+	inheritedRoleTemplateName := namegen.AppendRandomString("customrole")
+	inheritedRoleTemplate, err := grw.client.Management.RoleTemplate.Create(&management.RoleTemplate{
+		Context:         projectContext,
+		Name:            inheritedRoleTemplateName,
+		RoleTemplateIDs: []string{roleProjectOwner},
+	})
+	require.NoError(grw.T(), err)
+
+	_, err = createGlobalRole(grw.client, []string{inheritedRoleTemplate.ID})
+	require.Error(grw.T(), err)
+
+	expectedErrMessage := "admission webhook \"rancher.cattle.io.globalroles.management.cattle.io\" denied the request: globalrole.inheritedClusterRoles: Invalid value: \"" + inheritedRoleTemplate.ID + "\": unable to bind a roleTemplate with non-cluster context: project"
+
+	start := "admission webhook"
+	end := "Unable to bind a roleTemplate with non-cluster context: project"
+
+	getError := getErrorMessage(err.Error(), start, end)
+
+	assert.Contains(grw.T(), expectedErrMessage, getError)
+
+}
+
+func (grw *GlobalRolesV2WebhookTestSuite) TestRoleTemplateWithBadUserSubject() {
+
+	log.Info("Verify creating a cluster role template binding with the label grb Owner for a new user is rejected by webhook")
+	user, err := createGlobalRoleAndUser(grw.T(), grw.client, []string{roleOwner})
+	require.NoError(grw.T(), err)
+
+	crtbList, err, grbOwner := getCRTBFromGRBOwner(grw.T(), grw.client, user, grw.clusterCount)
+	require.NoError(grw.T(), err)
+
+	for _, crtb := range crtbList.Items {
+		log.Info("Create a new user with global role standard user and custom global role.")
+
+		createdUser, err := users.CreateUserWithRole(grw.client, users.UserConfig(), standardUser, globalRole.Name)
+		require.NoError(grw.T(), err)
+
+		crtbName := namegen.AppendRandomString("test-")
+		principalID := localPrefix + createdUser.Name
+		createdCRTB, err := rbac.CreateClusterRoleTemplateBinding(grw.client, localcluster, crtb.Namespace, crtbName, createdUser.Name, crtb.RoleTemplateName, principalID, crtb.Annotations, crtb.Labels)
+		require.NoError(grw.T(), err)
```

Comment:
Do we expect an error here since we do not expect crtb creation to succeed?

Reply:
No, it won't error out. CRTB creation will succeed but is immediately deleted by the controllers.